### ISSUE TITLE
HTTP -> HTTPS redirect returns a 404

### ISSUE THREAD

**Issue by JonasKs** [containous/traefik#7136]
### Do you want to request a *feature* or report a *bug*?

Bug (I assume)

### What did you do?

Set up a simple `traefik.yml`:

```yaml
log:
  level: DEBUG
api:
  dashboard: true
  insecure: true
entryPoints:
  proxy:
    address: :8083
providers:
  file:
    directory: /etc/traefik/dynamic-config

```

where I have the dynamic-config:
```yml
http:
  routers:
    test:
      rule: Host(`my-route.domain.com`)
      service: test
      tls:
        certificates:
          - certFile: /etc/traefik/companyroot.crt
  services:
    test:
      loadBalancer:
        servers:
          - url: https://192.168.0.1
            scheme: https
```

The pod is hosted in an Openshift cluster. I'm accessing `https://my-route.domain.com/test` but I always get a `404 page not found` from traefik. 

If I remove the TLS lines and have a config like this:
```yml
http:
  routers:
    test:
      rule: Host(`my-route.domain.com`)
      service: test
  services:
    test:
      loadBalancer:
        servers:
          - url: https://192.168.0.0
            scheme: https
```

I get a `Internal Server Error`, with the log messages:
```bash
time = "2020-08-03T08:39:16Z"
level = debug
msg = "vulcand/oxy/roundrobin/rr: begin ServeHttp on request"
Request = {"Method": "GET",
           "URL": {"Scheme": "", "Opaque": "", "User": null, "Host": "", "Path": "/test",
                   "RawPath": "", "ForceQuery": false, "RawQuery": "", "Fragment": ""}, "Proto": "HTTP/1.1",
           "ProtoMajor": 1, "ProtoMinor": 1,
           "Header": {"Accept": ["text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8"],
                      "Accept-Encoding": ["gzip, deflate, br"], "Accept-Language": ["en-US,en;q=0.5"],
                      "Cache-Control": ["max-age=0"], "Cookie": ["experimentation_subject_id=..."],
                      "Forwarded": ["for=10.73.13.2;host=my-route.domain.com;proto=https;proto-version="],
                      "Upgrade-Insecure-Requests": ["1"],
                      "User-Agent": ["Mozilla/5.0 (X11; Linux x86_64; rv:79.0) Gecko/20100101 Firefox/79.0"],
                      "X-Forwarded-Host": ["my-route.domain.com"], "X-Forwarded-Port": ["80"],
                      "X-Forwarded-Proto": ["http"], "X-Forwarded-Server": ["traefik-21-ctl2l"],
                      "X-Real-Ip": ["10.128.0.1"]}, "ContentLength": 0, "TransferEncoding": null,
           "Host": "my-route.domain.com", "Form": null, "PostForm": null, "MultipartForm": null, "Trailer": null,
           "RemoteAddr": "10.128.0.1:56368", "RequestURI": "/test", "TLS": null}
time = "2020-08-03T08:39:16Z"
level = debug
msg = "vulcand/oxy/roundrobin/rr: Forwarding this request to URL"
Request = {"Method": "GET",
           "URL": {"Scheme": "", "Opaque": "", "User": null, "Host": "", "Path": "/test",
                   "RawPath": "", "ForceQuery": false, "RawQuery": "", "Fragment": ""}, "Proto": "HTTP/1.1",
           "ProtoMajor": 1, "ProtoMinor": 1,
           "Header": {"Accept": ["text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8"],
                      "Accept-Encoding": ["gzip, deflate, br"], "Accept-Language": ["en-US,en;q=0.5"],
                      "Cache-Control": ["max-age=0"], "Cookie": ["experimentation_subject_id=..."],
                      "Forwarded": ["for=10.73.13.2;host=my-route.domain.com;proto=https;proto-version="],
                      "Upgrade-Insecure-Requests": ["1"],
                      "User-Agent": ["Mozilla/5.0 (X11; Linux x86_64; rv:79.0) Gecko/20100101 Firefox/79.0"],
                      "X-Forwarded-Host": ["my-route.domain.com"], "X-Forwarded-Port": ["80"],
                      "X-Forwarded-Proto": ["http"], "X-Forwarded-Server": ["traefik-21-ctl2l"],
                      "X-Real-Ip": ["10.128.0.1"]}, "ContentLength": 0, "TransferEncoding": null,
           "Host": "my-route.domain.com", "Form": null, "PostForm": null, "MultipartForm": null, "Trailer": null,
           "RemoteAddr": "10.128.0.1:56368", "RequestURI": "/test", "TLS": null}
ForwardURL = "https://192.168.0.1 "
time = "2020-08-03T08:39:18Z"
level = debug
msg = "'500 Internal Server Error' caused by: x509: certificate signed by unknown authority"
time = "2020-08-03T08:39:18Z"
level = debug
msg = "vulcand/oxy/roundrobin/rr: completed ServeHttp on request"
Request = {"Method": "GET",
           "URL": {"Scheme": "", "Opaque": "", "User": null, "Host": "", "Path": "/test",
                   "RawPath": "", "ForceQuery": false, "RawQuery": "", "Fragment": ""}, "Proto": "HTTP/1.1",
           "ProtoMajor": 1, "ProtoMinor": 1,
           "Header": {"Accept": ["text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8"],
                      "Accept-Encoding": ["gzip, deflate, br"], "Accept-Language": ["en-US,en;q=0.5"],
                      "Cache-Control": ["max-age=0"], "Cookie": ["experimentation_subject_id=..."],
                      "Forwarded": ["for=10.73.13.2;host=my-route.domain.com;proto=https;proto-version="],
                      "Upgrade-Insecure-Requests": ["1"],
                      "User-Agent": ["Mozilla/5.0 (X11; Linux x86_64; rv:79.0) Gecko/20100101 Firefox/79.0"],
                      "X-Forwarded-Host": ["my-route.domain.com"], "X-Forwarded-Port": ["80"],
                      "X-Forwarded-Proto": ["http"], "X-Forwarded-Server": ["traefik-21-ctl2l"],
                      "X-Real-Ip": ["10.128.0.1"]}, "ContentLength": 0, "TransferEncoding": null,
           "Host": "my-route.domain.com", "Form": null, "PostForm": null, "MultipartForm": null, "Trailer": null,
           "RemoteAddr": "10.128.0.1:56368", "RequestURI": "/test", "TLS": null}

```
So, I assume there is _some_ bug, since I get **0** log messages when I enable TLS, and without TLS I get a internal server error. 

The 192.168.0.1 address is actually an external web service, not in the cluster. 

### What did you expect to see?
Fetching `https://my-route.domain.com/test` would actually fetch `https://192.168.0.1/test`

### What did you see instead?
`404 page not found`

### Output of `traefik version`: (_What version of Traefik are you using?_)

```bash
Version:      2.2.8
Codename:     chevrotin
Go version:   go1.14.6
Built:        2020-07-28T15:46:03Z
OS/Arch:      linux/amd64
```

Fetched from `FROM traefik:2.2.8`, copied over files. 

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Hosted in OpenShift. It's a bunch of `kustomization.yml` files, but I can merge them together and provide an example if you need it. 

TL;DR: 
* Exposed containerports `8080` and `8083`
* One route goes to `8080`, and give me the dashboard. (This works) 
* The other route (named `my-route.domain.com` in this issue) goes to `8083`. 

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)
See above.

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of communication regulation.

Thanks for your interest in Traefik!

We dedicate the issue tracker to bug reports and feature requests only. My advanced AI has spotted that your issue might be a configuration problem or relates to something that doesn't look like a bug.

To confirm this, please join our [Community Forum](https://community.containo.us/) and reach out to us on the [Traefik section](https://community.containo.us/c/traefik).

In case I'm wrong (well, that would be embarrassing :sweat_smile:), my developers will re-open the issue and fix me!

In the meantime, you can double check [Traefik's documentation](https://docs.traefik.io/).
